frame/support/src/lib.rs

```diff
@@ -210,6 +210,29 @@ impl TypeId for PalletId {
 /// ```
 pub use frame_support_procedural::storage_alias;
 
+/// A series of attribute macros related to benchmarking. Right now just
+/// [`benchmarking(cached)`](frame_support_procedural::benchmarking(cached)) is provided.
+///
+/// # benchmarking(cached)
+///
+/// This is an attribute macro that can be attached to a [`StorageValue`]. Doing
+/// so will exclude reads of that value's storage key from counting towards
+/// weight calculations during benchmarking and will also ensure that the value
+/// is automatically cached/read at the beginning of every block.
```

Comment:
I dont think that the value being cached is actually ensured yet, that would require us to explicitly read each storage item in the beginning of the block.

I think we talked about that being a follow up PR, but happy to have it here too.

Reply:
Yeah I'll update the comment this is based on an old understanding of the issue. We'll do this in a subsequent PR